### DISCUSSION TITLE
WalletConnectUnity

### DISCUSSION THREAD

**Discussion by laureveler** [WalletConnect/walletconnect-monorepo#3284] [Developer Support]
I'm experimenting with the Demo DApp and have encountered a few issues:

1) How to display wallet balance following connection? The example code to do this has been commented out and attempting to use it results in a JSON parse error (can't convert BigInteger to string).
EDIT - Resolved

```
 var request = new EthGetBalance(address);

            var resultTask = _wc.Request<EthGetBalance, BigInteger>(currentSession.Topic, request, chainId);

            yield return new WaitForTaskResult<BigInteger>(resultTask);

            var result = resultTask.Result;
```
(If anyone else gets stuck here - change the BigInteger into a string. The result you get must be converted from hexadecimal and divided by 1000000000000000000).

2) Sign transaction demo -

When attemping a transaction I get the message 
`ArgumentException: The module Wallet Connect Test-core-history-of-type-SessionRequest`1 with context Wallet Connect Test-core-history-of-type-SessionRequest`1 is attempting to create a new EventDelegator that overlaps with an existing EventDelegator`
Any help appreciated

**Comment by underclockeddev** [comment]
Second on this EventDelegator question. I've found that changing the Project Name and Base Context to something completely unique on every single session prevents this exception, but I'm not sure why that fixes it.

**Reply by underclockeddev** [reply]
Created an issue because this issue persists for me https://github.com/WalletConnect/WalletConnectUnity/issues/114
